tests/TestCase/ORM/QueryTest.php

```diff
@@ -3576,6 +3578,29 @@ public function testInnerJoinWithSelect()
         $this->assertEquals($expected, $results);
     }
 
+    /**
+     * Tests contain() in query returned by innerJoinWith throws exception.
+     *
+     * @return void
+     */
+    public function testInnerJoinWithContain()
+    {
+        $comments = $this->getTableLocator()->get('Comments');
+        $articles = $comments->belongsTo('Articles');
+        $articles->hasOne('ArticlesTranslations');
+
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('Cannot contain associations in query returned by query builder');
+        $comments
+            ->find()
+            ->innerJoinWith('Articles', function ($q) {
```

Comment:
Can you please typehint $q here, as the ->contain() and ->where() are not detected right now (not followable).